src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_helpers.cc

```diff
@@ -291,21 +291,25 @@ smart_objects::SmartObject RCHelpers::MergeModuleData(
 
   smart_objects::SmartObject result = data1;
 
-  for (auto it = data2.map_begin(); it != data2.map_end(); ++it) {
-    const std::string& key = it->first;
-    smart_objects::SmartObject& value = it->second;
-    if (!result.keyExists(key) || value.getType() != result[key].getType()) {
+  if (!data2.empty()) {
+    for (auto it = data2.map_begin(); it != data2.map_end(); ++it) {
+      const std::string& key = it->first;
+      smart_objects::SmartObject& value = it->second;
+      if (!result.keyExists(key) || value.getType() != result[key].getType()) {
+        result[key] = value;
+        continue;
+      }
+
+      // Merge maps and arrays with `id` param included, replace other types
+      if (value.getType() == smart_objects::SmartType::SmartType_Map) {
```

Comment:
Please swap comparables. Also please check similar cases in PR.

Reply:
@YarikMamykin Done in 9c5267d33e71ca65d7ebf61ba4058e2f6fd4b2eb